tests/integration_tests/test_ingest_component_pathfinders.py

```diff
@@ -0,0 +1,140 @@
+            "auth": json.dumps(
+                {
+                    "Programme": [
+                        "Lewes District Council",
+                    ],
+                    "Fund Types": [
+                        "Pathfinders",
+                    ],
+                }
+            ),
+            "do_load": False,
+        },
+    )
+
+    assert response.status_code == 400, f"{response.json}"
+    assert response.json["detail"] == "Workbook validation failed"
+    assert len(response.json["pre_transformation_errors"]) == 1
+
+    assert (
+        "You’re not authorised to submit for Rotherham Metropolitan Borough Council. You can only submit for Lewes"
+        " District Council."
+    ) in response.json["pre_transformation_errors"]
+
+
+def test_ingest_pf_r1_basic_errors(test_client, pathfinders_round_1_file_failure):
+    """Tests that, with incorrect values present in Excel file, the endpoint returns initial validation errors."""
+    endpoint = "/ingest"
+    response = test_client.post(
+        endpoint,
+        data={
+            "excel_file": pathfinders_round_1_file_failure,
+            "fund_name": "Pathfinders",
+            "reporting_round": 1,
+            "auth": json.dumps(
+                {
+                    "Programme": [
+                        "Rotherham Metropolitan Borough Council",
+                    ],
+                    "Fund Types": [
+                        "Pathfinders",
+                    ],
+                }
+            ),
+            "do_load": False,
+        },
+    )
+
+    assert response.status_code == 400, f"{response.json}"
+    assert response.json["detail"] == "Workbook validation failed"
+    assert len(response.json["pre_transformation_errors"]) == 2
+
+    assert "The expected reporting period is Q3 Oct - Dec 23/24" in response.json["pre_transformation_errors"]
+    assert "The expected value is V 1.0" in response.json["pre_transformation_errors"]
+
+
+def test_ingest_pf_incorrect_round(test_client, pathfinders_round_1_file_success):
+    """Tests that, with an incorrect reporting round, the endpoint throws an unhandled exception."""
+    # TODO is this the desired behaviour?
+    with pytest.raises(ValueError) as e:
+        endpoint = "/ingest"
+        test_client.post(
+            endpoint,
+            data={
+                "excel_file": pathfinders_round_1_file_success,
+                "fund_name": "Pathfinders",
+                "reporting_round": 2,
+                "auth": json.dumps(
+                    {
+                        "Programme": [
+                            "Rotherham Metropolitan Borough Council",
+                        ],
+                        "Fund Types": [
+                            "Pathfinders",
+                        ],
+                    }
+                ),
+                "do_load": False,
+            },
+        )
+
+    assert e.value.args == ("There are no IngestDependencies for Pathfinders round 2",)
```

Comment:
Can we assert against a 500 response here instead of catching an `Exception` as these tests should be asserting against the response rather than anything at the Python level

Reply:
Thinking about this more.. I don't think it makes sense to return a 500 error here. We should be returning a 400 (bad request) response instead. 

It would be best to change `ingest_dependencies_factory` to return `None` in the case that fund and reporting round are invalid here: 
https://github.com/communitiesuk/funding-service-design-post-award-data-store/blob/682cce7c645b2fa6cd5118f819da4b4976daaded/core/controllers/ingest_dependencies.py#L88-L89

and then `controllers.ingest.ingest` can handle the `None` by returning a 400 response here:
https://github.com/communitiesuk/funding-service-design-post-award-data-store/blob/682cce7c645b2fa6cd5118f819da4b4976daaded/core/controllers/ingest.py#L78